wkconnect/routes/datasets/histogram.py

```diff
@@ -0,0 +1,110 @@
+import asyncio
+import time
+from collections import OrderedDict
+from typing import List
+
+import numpy as np
+from sanic import Blueprint, response
+from sanic.request import Request
+
+from ...utils.json import to_json
+from ...utils.types import Vec3D
+from ...webknossos.access import AccessRequest, authorized
+from ...webknossos.models import BoundingBox as WkBoundingBox
+from ...webknossos.models import DataRequest as WKDataRequest
+from ...webknossos.models import Histogram
+
+histogram = Blueprint(__name__)
+
+
+@histogram.route(
+    "/<organization_name>/<dataset_name>/layers/<layer_name>/histogram", methods=["GET"]
+)
+@authorized(AccessRequest.read_dataset)
+async def histogram_post(
+    request: Request, organization_name: str, dataset_name: str, layer_name: str
+) -> response.HTTPResponse:
+    (backend_name, dataset) = request.app.repository.get_dataset(
+        organization_name, dataset_name
+    )
+    backend = request.app.backends[backend_name]
+
+    layer = [i for i in dataset.to_webknossos().dataLayers if i.name == layer_name][0]
+
+    sample_positions = generate_sample_positions(2, layer.boundingBox, 32)
+    sample_positions_distinct = list(OrderedDict.fromkeys(sample_positions))
+
+    bucket_requests = [
+        WKDataRequest(position, zoomStep=1, cubeSize=32, fourBit=False)
+        for position in sample_positions_distinct
+    ]
+
+    before = time.time()
+    buckets = await asyncio.gather(
+        *(
+            backend.read_data(
+                dataset,
+                layer_name,
+                r.zoomStep,
+                Vec3D(*r.position),
+                Vec3D(r.cubeSize, r.cubeSize, r.cubeSize),
+            )
+            for r in bucket_requests
+        )
+    )
+    existing_buckets = [
+        data for r, data in zip(bucket_requests, buckets) if data is not None
+    ]
+    data = np.concatenate(existing_buckets) if len(existing_buckets) > 0 else b""
+    data = data[np.nonzero(data)]
+
+    after = time.time()
+    print(
+        f"Fetched histogram data for {len(sample_positions_distinct)} positions, shape {data.shape} dtype {data.dtype}. Took {(after - before)*1000:.2f} ms."
+    )
+
+    if np.issubdtype(data.dtype, np.integer):
+        minimum, maximum = np.iinfo(data.dtype).min, np.iinfo(data.dtype).max
+        counts, _ = np.histogram(data, bins=np.arange(maximum))
```

Comment:
Wouldn't it be enough if we use the bins from minimum to maximum?

Reply:
If the minimum is always `0` since the dtype should be a uint we should assert it here.